packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -375,23 +375,31 @@ class RenderEditable extends RenderBox {
     return _textPainter.maxIntrinsicWidth;
   }
 
-  // This does not required the layout to be updated.
-  double get _preferredLineHeight => _textPainter.preferredLineHeight;
+  /// The height of a typical text line in logical pixels.
+  ///
+  /// This height is based on properties of the text's style such as
+  /// font size etc and could change when those properties change.
```

Comment:
text and textAlign are upstream of the painter. You set them. As far as I can tell, this is the first getter that would expose layout-based data.

Generally, this is an anti-pattern. If another object reads the layout, e.g. if the parent render object uses the preferredLineHeight in some way, then this render object changes its configuration, the parent won't update, until randomly the other child is dirtied, then suddenly it will update. We go to extreme lengths to make it possible to read the size, the intrinsic dimensions, etc, in such a way that the right objects get dirtied when the values change to avoid these problems.

I don't see anything in this file which causes the EditableTextState to be updated when the layout changes.